src/compiler/checker.ts

```diff
@@ -18908,6 +18908,9 @@ namespace ts {
                 target = (<BinaryExpression>target).left;
             }
             if (target.kind === SyntaxKind.ObjectLiteralExpression) {
+                if (strictNullChecks && (<ObjectLiteralExpression>target).properties.length === 0) {

```

Comment:
I would move this code into `checkObjectLiteralAssignment`.

Reply:
Done!